src/components/Snippet/index.tsx

```diff
@@ -0,0 +1,238 @@
+  width: ${({ prefix }) => (prefix === 'lines' ? '3%' : '')};
+  display: inline-flex;
+  justify-content: flex-end;
+  text-align: right;
+  user-select: none;
+  margin-right: ${({ theme }) => theme.space['1']};
+`
+
+const Container = styled('div', {
+  shouldForwardProp: prop => !['multiline'].includes(prop),
+})<{ multiline?: boolean }>`
+  position: relative;
+  display: flex;
+  justify-content: start;
+  max-width: 100%;
+  ${({ multiline }) => (multiline ? 'width: 100%;' : '')}
+  background: ${({ theme }) => theme.colors.neutral.backgroundStrong};
+  border-radius: ${({ theme }) => theme.radii.default};
+`
+
+const StyledStack = styled(Stack)`
+  width: 100%;
+`
+
+const ButtonContainer = styled('div', {
+  shouldForwardProp: prop => !['multiline'].includes(prop),
+})<{ multiline?: boolean }>`
+  position: absolute;
+  top: 0;
+  right: 0;
+  padding: ${({ theme, multiline }) =>
+    multiline
+      ? `${theme.space['2']} ${theme.space['2']} 0 0`
+      : `${theme.space['1']} ${theme.space['2']} 0 0`};
+  background: ${({ theme }) => theme.colors.neutral.backgroundStrong};
+  border-radius: ${({ theme }) => theme.radii.default};
+  border: 2px solid transparent;
+  ${({ multiline, theme }) =>
+    !multiline
+      ? `box-shadow: -27px 0 19px -11px ${theme.colors.neutral.backgroundStrong}`
+      : ''};
+`
+
+const ShowMoreContainer = styled('div', {
+  shouldForwardProp: prop => !['showMore'].includes(prop),
+})<{ showMore?: boolean }>`
+  width: 100%;
+  box-shadow: ${({ theme, showMore }) =>
+    !showMore
+      ? `0px -22px 19px -6px
+    ${theme.colors.neutral.backgroundStrong}`
+      : 'none'};
+`
+
+const StyledButton = styled.button`
+  width: 100%;
+  background: none;
+  border: none;
+  padding: ${({ theme }) => theme.space['2']};
+  padding-top: ${({ theme }) => theme.space['1']};
+  cursor: pointer;
+`
+
+const AlignCenterText = styled(Text)`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`
+
+const AnimatedArrowIcon = styled(Icon, {
+  shouldForwardProp: prop => !['showMore'].includes(prop),
+})<{ showMore?: boolean }>`
+  transform: ${({ showMore }) =>
+    showMore ? 'rotate(180deg)' : 'rotate(0deg)'};
+  transform-origin: center;
+  transition: transform 300ms ease-in-out;
+`
+
+type SnippetProps = {
+  value: string
+  multiline?: boolean
+  /**
+   * prefix display an element at the beginning of the snippet that is not copiable or selectable.
+   * For `lines` prefix it will display the line number.
+   * For `command` prefix it will display a `$` sign.
+   */
+  prefix?: Prefixes
+  showText?: string
+  hideText?: string
+} & Pick<CopyButtonProps, 'copyText' | 'copiedText'>
+
+const Snippet = ({
+  value,
+  copyText,
+  copiedText,
+  multiline = false,
+  showText = 'Show',
+  hideText = 'Hide',
+  prefix,
+}: SnippetProps) => {
+  const theme = useTheme()
+  const [showMore, setShowMore] = useState(false)
+
+  const lines = value.split(/\r\n|\r|\n/).filter(Boolean)
```

Comment:
Could we extract to a const so we can provide more context about what this regex is?

```suggestion
const LINE_BREAK_REGEX = /\r\n|\r|\n/
...
  const lines = value.split(LINE_BREAK_REGEX).filter(Boolean)
```